controller/controller.go

```diff
@@ -495,14 +495,10 @@ type permanentError struct {
 	e error
 }
 
-// IsPermanentError returns true if given error is permanentError
+// IsPermanentError returns true if the given error is a permanentError.
 func IsPermanentError(err error) bool {
-	switch err.(type) {
-	case permanentError:
-		return true
-	default:
-		return false
-	}
+	_, ok := err.(permanentError)
```

Comment:
can we use errors.Is instead?

Reply:
That's what I also expected, but if I change that function to
```diff
 func IsPermanentError(err error) bool {
-       _, ok := err.(permanentError)
-       return ok
+       return errors.Is(err, permanentError{})
 }
```
my test fails with `Expected type controller.permanentError to be a permanentError`.

Reply:
@vagababov oh, I know why. `Is()` calls `Unwrap()`, which returns the wrapped error instead of the permanentError.

Reply:
I see.
Thanks.

Reply:
Solved in accacc9f9a883fe0793fa3a7b78dc8b6bc9664c6.
It's actually a good pattern because a permanentError can be wrapped itself, and still be permanent.